### DISCUSSION TITLE
Add a command to change a player's bed location

### DISCUSSION THREAD

**Discussion by sakurawald** [Zrips/CMI#7249] [Feature Requests]
Add a command to change a player's bed location.
like /cmi setbed <player> to set current location to his bed location.
so the vanilla re-spawn will respawn the player to his bed location.

the command /cmi setspawn will change the location where one player issue command /spawn.
but i hope the players have the same one /spawn location, but i can set their bed location independently

**Comment by mrfloris** [comment]
Interesting idea, i wonder if this is technically even possible.

**Reply by sakurawald** [reply]
Yeah, some other plugin can set a newbie player's bed-location. I mean BetterRTP plugin.
What i expetced is:
we'll random teleport a newbie-player, and then set the player's bed-location.
so that when the player is died, he will respawn at his bed-location (we use the vanilla re-spawn), instead always respawn in spawn (where /spawn command teleport to)

**Comment by mrfloris** [comment]
i am not a proper dev, maybe i dont understand this, but how does the plugin know WHICH bed to use for player location if the player isn't touching the bed? 
If you RT a player, they end up in some random chunk with maybe 0 beds in it .. 

CMI has bed respawn priority, and touch bed for bed home .. these features are in cmi. 

but dunno how a plugin can rt a user and go 'ok bed location set', even though there's no bed, wouldn't respawn just say 'bed obstructed or broken' and reset them to spawn ?

**Reply by sakurawald** [reply]
> 

I'd like to say. Every player has his own file to save his playerdata.
the playerdata contains the player's inventory, ender-chest items, and his re-spawn location.
so it's possible to only change one player's respawn location. just like the vanilla bed-location.
every player can set his own bed and respawn at his bed.

**Reply by mrfloris** [reply]
If a player is at spawn and types /rtp and ends up in /survival at XYZ, and dies, their respawn is at spawn, unless cmi is told that the current world is priority, then it goes to survival's spawn point. 

if the player is at spawn and types /rtp and ends up in /survival at XYZ, walks around and hits a bed, and bed respawn is set in cmi as respawn priority, and they die, they will end up at the bed,  if the bed isn't found, the first situation is what will happen. 

CMI doesn't have a file for each player btw, they are all stored in a database (the .db file), back, deathback, pos, respawnpos, etc are all logged there. 

cmi's config.yml has per world respawn priority, you can set spawn for world respawn, you can set survival for bed first, then world then spawn world, you can set nether for anchor, then bed, then world, then regular spawn  etc.

**Reply by sakurawald** [reply]
yeah, the playerdata is the vanilla file. spigot apis may contains a method to set this.
